snakemake_executor_plugin_slurm/__init__.py

```diff
@@ -83,6 +123,38 @@ class ExecutorSettings(ExecutorSettingsBase):
 # Required:
 # Implementation of your executor
 class Executor(RemoteExecutor):
+    def clean_old_logs(self) -> None:
+        """Delete files older than specified age from the SLURM log directory.
+
+        Args:
+            logdir: Path to the log directory
+            age_cutoff: Number of days after which files should be deleted
+        """
+        # shorthands:
+        age_cutoff = self.workflow.executor_settings.delete_logfiles_older_than
+        logdir = self.workflow.executor_settings.logdir
+        keep_all = self.workflow.executor_settings.keep_successful_logs
+        if age_cutoff <= 0 or keep_all:
+            return
+        cutoff_secs = age_cutoff * 86400
+        current_time = time.time()
+        self.logger.info(f"Cleaning up log files older than {age_cutoff} day(s)")
+        for root, _, files in os.walk(logdir, topdown=False):
+            for fname in files:
+                file_path = os.path.join(root, fname)
+                try:
+                    file_age = current_time - os.stat(file_path).st_mtime
+                    if file_age > cutoff_secs:
+                        os.remove(file_path)
+                except (OSError, FileNotFoundError) as e:
+                    self.logger.warning(f"Could not delete file {file_path}: {e}")
+            # remove empty rule top dir, if empty
+            try:
+                if len(os.listdir(root)) == 0:
+                    os.rmdir(root)
+            except (OSError, FileNotFoundError) as e:
+                self.logger.warning(f"Could not remove empty directory {root}: {e}")
```

Comment:
@coderabbitai, can you suggest a rewrite of this using pathlib?

Reply:
will be tested

Reply:
done - with minor adjustments - see last push.